src/main/java/org/tugraz/sysds/runtime/matrix/data/LibMatrixReorg.java

```diff
@@ -2341,4 +2455,87 @@ public Long call() {
 			return rexpandColumns(_in, _out, _max, _cast, _ignore, _rl, _ru);
 		}
 	}
+
+	private static class SortTask implements Callable<Object>
+	{
+		private final int _start;
+		private final int _end;
+		private final int[] _indexes;
+		private final double[] _values;
+
+
+		protected SortTask(int start, int end, int[] indexes, double[] values){
+			_start = start;
+			_end = end;
+			_indexes = indexes;
+			_values = values;
+			
+		}
+
+		@Override
+		public Long call(){
+			SortUtils.sortByValue(_start, _end, _values, _indexes);
+			return 1l;
+		}
+
+	}
+
+	private static class MergeTask implements Callable<Object>
+	{
+		private final int _start;
+		private final int _end;
+		private final int _blockSize;
+		private final int[] _indexes;
+		private final double[] _values;
+
+		protected MergeTask(int start, int end, int blockSize, int[] indexes, double[] values){
+			_start = start;
+			_end = end;
+			_blockSize = blockSize;
+			_indexes = indexes;
+			_values = values;
+		}
+
+		@Override
+		public Long call(){
+			//Node node = new Node();
```

Comment:
That seems to be a remainder of previous code?

Reply:
Yes, I tried to create a linked list, and that performed worse. 
See https://github.com/tugraz-isds/systemds/pull/65/commits/1e3d8f1375fbc6d3058c8bc78ed1a73fde8429ac